src/main/java/gift/repository/GiftRepository.java

```diff
@@ -20,39 +22,58 @@ public class GiftRepository {
     // 하지만 생성자가 아닌 필드측에서 ConcurrentMap 을 생성시키면, GiftRepository 에 대해서 giftMap 에 대한
     // 접근을 할 수 있도록 하는 것이므로 데이터에 대한 통일을 이뤄낼 수 있다.
     private final Map<Long, Gift> giftMap = new ConcurrentHashMap<>();
+    private final static AtomicLong giftId = new AtomicLong(1);
 
     public GiftRepository() {}
 
-    public void save(RequestGift requestGift) {
-        giftMap.put(requestGift.id(), RequestGift.from(requestGift));
+    public ResponseGift save(RequestGift requestGift) {
+        giftMap.put(giftId.get(), RequestGift.toEntity(requestGift));
+        increaseId();
+        return ResponseGift.toJSON(giftId.get()-1, RequestGift.toEntity(requestGift));
     }
 
-    public Gift findById(Long id) {
+    public ResponseGift findById(Long id) {
         Gift gift = giftMap.get(id);
         if(gift == null){
             throw new RuntimeException("해당 상품이 없습니다.");
         }
-        return giftMap.get(id);
+        return ResponseGift.toJSON(id, giftMap.get(id));
     }
 
-    public List<Gift> findAll(){
-        return new ArrayList<>(giftMap.values());
+    public List<ResponseGift> findAll(){
+        List<ResponseGift> responseGifts = new ArrayList<>();
+        for(Long id : giftMap.keySet()){
+            responseGifts.add(ResponseGift.toJSON(id, giftMap.get(id)));
+        }
+        return responseGifts;
```

Comment:
```suggestion
        return giftMap
                .entrySet()
                .stream()
                .map((entry) -> ResponseGift.toJSON(entry.getKey(), entry.getValue()))
                .toList();
```

java의 stream api를 활용하면 이렇게 한 줄로도 작성이 가능해요!
사용법에 익숙해지면 코드 작성도 편해지고 가독성도 좋아지니 학습해서 사용하시는 것을 추천드려요~